diff -ur a/luci/public/js/busy.js b/luci/public/js/busy.js
--- a/luci/public/js/busy.js	2012-01-23 11:20:32.000000000 -0500
+++ b/luci/public/js/busy.js	2013-07-18 00:32:34.881665811 -0400
@@ -65,7 +65,8 @@
                         if (cur.db_task_type != 2 && cur.db_task_type != 9) {
                             // 2=set conf, 9=activate conf
                             try {
-                                $.jnotify(cur.db_status + ' completed successfully', true);
+								$('<div class="info">' + cur.db_status + " completed successfully" + '</div>').appendTo($('#flash'));
+                                //$.jnotify(cur.db_status + ' completed successfully', true);
                             } catch (e) {}
                         }
                         if (cur.db_task_type == 1 || cur.db_task_type == 3 ||
@@ -94,7 +95,8 @@
                     $('<li class="status_msg_text">' + cur.db_status + substatus + '</li>').appendTo($('#status_msg_details'));
                 } else if (cur.ricci_code == -1) {
                     try {
-                        $.jnotify(cur.db_status + " failed: " + cur.ricci_msg, "error", true);
+						$('<div class="error">' + cur.db_status + " failed: " + cur.ricci_msg + '</div>').appendTo($('#flash'));
+                        //$.jnotify(cur.db_status + " failed: " + cur.ricci_msg, "error", true);
                         cur_errors++;
                     } catch (e) {}
                 } else {
diff -ur a/luci/templates/master.html b/luci/templates/master.html
--- a/luci/templates/master.html	2013-07-18 00:32:25.120904586 -0400
+++ b/luci/templates/master.html	2013-07-18 00:33:17.964612076 -0400
@@ -58,6 +58,7 @@
   <div id="content">
     <py:with vars="flash=tg.flash_obj.render('flash', use_js=False)">
       <div py:if="flash" py:content="XML(flash)" class="message"/>
+      <div py:if="not flash" id="flash"/>
     </py:with>
     <py:if test="'stern_warning' in dir(tmpl_context) and tmpl_context.stern_warning">
       <div id="stern_warning" title="Warning">
